psydac/api/ast/nodes.py

```diff
@@ -131,17 +134,29 @@ def rhs(self):
 #==============================================================================
 class IndexNode(Expr):
     """Base class representing one index of an iterator"""
-    def __new__(cls, length=None):
+    def __new__(cls, start=0, stop=None, length=None):
         obj = Basic.__new__(cls)
+        obj._start  = start
+        obj._stop   = stop
         obj._length = length
         return obj
 
+    @property
+    def start(self):
+        return self._start
+
+    @property
+    def stop(self):
+        return self._stop
+
     @property
     def length(self):
         return self._length
 
-    def set_length(self, length):
-        obj = type(self)(length)
+    def set_range(self, start=Tuple(*[S.Zero]*3), stop=None, length=None):
```

Comment:
Couldn't you use `start=0` as a default value? This is also the default value of the constructor.

Reply:
I understand that Sympde and Psydac are not going to work in 4D at the moment. But most of Psydac is written to be dimension-agnostic and hence it allows any number of dimensions, so I would like as much new code as possible to be flexible as well.

But couldn't you use `start=0` as the default value? This is also the default value of the constructor.

Reply:
okay I will use that

Reply:
mmm, this will make things more difficult in the loops, so I will create another object just for that, I will call it `TensorInteger` , that returns a tuple of the right dimension later on in the code

Reply:
done